### ISSUE TITLE
stable name for execution root convenience symlink

### ISSUE THREAD

**Issue by adam-azarchs** [bazelbuild/bazel#20799]
### Description of the feature request:

By default, bazel will put a symlink `bazel-$(basename $(dirname WORKSPACE))` in the workspace root pointing to `$(bazel info execution_root)`.  It's possible to modify the prefix (`bazel-`) on that with [`--symlink_prefix`](https://bazel.build/reference/command-line-reference#flag--symlink_prefix), but it's not possible to change the suffix.

A few ways to implement this, any of which would meet our use cases:
1. set the full name of the convenience symlink e.g. in `.bazelrc`
2. override the automatically-detected workspace basename used along with the prefix, again in `.bazelrc`
3. Use the workspace name as [declared](https://bazel.build/rules/lib/globals/workspace#workspace) in the `WORKSPACE` file, rather than the name of the directory that the code was checked out into.

### What underlying problem are you trying to solve with this feature?

The current scheme poses difficulties in places where we'd like to integrate bazel with other tools.  For example, one of our use cases is we'd like to be able use a `.vscode/c_cpp_properties.json` file into the repository to tell the editor about various include paths from downloaded remote repos, e.g. `bazel-src/external/libtiff/libtiff`, and we'd like to be able to check that file in to make onboarding more pleasant, as well as to make life more pleasant for people who might have multiple worktrees checked out in different locations.  The problem is that `bazel-src` there only works if you've checked the repository out in a directory named `src`.

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

release 6.3.2

**Comment by Wyverald** [comment]
Moving to team-Core for now, since I'm not really sure -- who owns those symlinks? (Feel free to assign back if it's not Core) Anyway, I think it would actually be nicer to make the `bazel-$DIRNAME` symlink have a static name instead... having it change based on whatever criteria seems like an odd choice in the first place.

**Comment by adam-azarchs** [comment]
Yes, just setting a constant name (e.g. `bazel-src`) would also work, though it would potentially be a breaking change.  Maybe that's fine, since most use cases for the symlink as is are going to be quite fragile to begin with.  But it we did want to preserve backwards compatibility, then either we'd want the suffix choice to be something that could be controlled by the repository (e.g. WORKSPACE or .bazelrc) or, maybe more simply, just have an `--incompatible` flag to change the behavior between the legacy behavior and just using a hard-coded suffix like `src`.

**Comment by kalvdans** [comment]
I've opened a PR to remove this convenience symlink, see https://github.com/bazelbuild/bazel/pull/29893 (actual change https://github.com/kalvdans/bazel/commit/198add5e76d07afcc5e5de80883cf6394da11f31 since pull request was closed by stale bots)
Please suggest how you would like the deprecation period to happen!
